server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -326,7 +326,7 @@ private ClusterState applyRequest(ClusterState currentState, PutMappingClusterSt
     }
 
     public void putMapping(final PutMappingClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
-        clusterService.submitStateUpdateTask("put-mapping",
+        clusterService.submitStateUpdateTask("put-mapping " + getRequestIndex(request),
```

Comment:
```suggestion
        clusterService.submitStateUpdateTask("put-mapping " + Strings.arrayToCommaDelimitedString(request.indices())),
```

This removes the need for the method below too.

Reply:
that's great